### PR TITLE
feat: better in-browser support

### PR DESCRIPTION
**Pull request by sentientwaffle**
* Don't try to import `URL` (which is a global in Node `>=10.0.0` and in the browser).
* Use WebCrypto to generate random bytes so that a crypto polyfill isn't needed.

### CODE REVIEW

**Comment by matdehaast** [comment]
@sentientwaffle Can you look to also replace the WS implementation with an isomorphic version? That was the other hurdle I ran into when trying to use this in the browser

File: src/browser/crypto.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+const { crypto } = window
+
+export function randomBytes (
+  size: number,
+  callback: (err: Error | null, buf: Buffer) => void
+): void {
+  const randArray = new Uint8Array(size)
+  const randValues = crypto.getRandomValues(randArray)
+  callback(null, Buffer.from(randValues))
```
Comment (DonChangfoot) [inline src/browser/crypto.ts:12]: @sentientwaffle I dont think Buffer is supported in the browser. Are we going to assume that people are going to be using webpack?

Comment (sentientwaffle) [reply src/browser/crypto.ts:12]: Yes, I'm assuming that it is built with webpack (or, in general, that all the necessary libraries are polyfilled).

**Review by matdehaast — APPROVED** [review]
Looks good. We don't have any test for the browser I suspect. Lets aim for another PR that tries to test/check the browser and also add documentation on how to use for a browser enviroment

**Pull request merged**